transforms/universal/html2parquet/python/src/html2parquet_main.py

```diff
@@ -0,0 +1,110 @@
+import time
+from argparse import ArgumentParser, Namespace
+from typing import Any
+import zipfile
+import io
+import trafilatura
+
+import pyarrow as pa
+# from data_processing_ray.runtime.ray import RayTransformLauncher
+# from data_processing_ray.runtime.ray.runtime_configuration import (
+#   RayTransformRuntimeConfiguration,
+# )
+# import data_processing
+# print(dir(data_processing))
+
+
+from data_processing.transform import AbstractTableTransform, AbstractBinaryTransform, TransformConfiguration
+from data_processing.utils import CLIArgumentProvider, get_logger, TransformUtils
+
+
+class HtmlToParquetTransform(AbstractTableTransform):
+    def __init__(self, config: dict[str, Any]):
+        self.sleep = config.get("sleep", 1)
+
+    def transform_binary(self, file_name: str, byte_array: bytes) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        """
+        Converts raw data file (ZIP) to Parquet format
+        """
+        # We currently only process .zip files
+        print("transform_binary")
+        if TransformUtils.get_file_extension(file_name)[1] != ".zip":
+            self.logger.warning(f"Got unsupported file type {file_name}, skipping")
+            return [], {}
+        data = []
+        number_of_rows = 0
+
+        with zipfile.ZipFile(io.BytesIO(bytes(byte_array))) as opened_zip:
+            # Loop through each file member in the ZIP archive
+            for member in opened_zip.infolist():
+                if not member.is_dir():
+                    with opened_zip.open(member) as file:
+                        try:
+                            # Read the content of the file
+                            content_bytes = file.read()
+                            #use Tra something library
+                            content_string = trafilatura.extract(content_bytes)
+                            print("content_string:", content_string)
+                            row_data = {
+                                "filename": member.filename,
+                                "contents": content_string
+                            }
```

Comment:
Are these the only 2 columns that we need? Pls take a look at pdf2parquet and code2parquet for ideas on additional columns